shell/src/main/java/alluxio/cli/validation/StorageSpaceValidationTask.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.util.FormatUtils;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Task for validating whether worker tiered storage has enough space.
+ */
+public final class StorageSpaceValidationTask implements ValidationTask {
+
+  /**
+   * Creates a new instance of {@link StorageSpaceValidationTask}
+   * for validating tiered storage space.
+   */
+  public StorageSpaceValidationTask() {
+  }
+
+  @Override
+  public boolean validate() {
+    int numLevel = Configuration.getInt(PropertyKey.WORKER_TIERED_STORE_LEVELS);
+    boolean success = true;
+
+    for (int level = 0; level < numLevel; level++) {
+      PropertyKey tierAliasConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_ALIAS.format(level);
+      String alias = Configuration.get(tierAliasConf);
+
+      PropertyKey tierDirPathConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_PATH.format(level);
+      String[] dirPaths = Configuration.get(tierDirPathConf).split(",");
+
+      PropertyKey tierDirCapacityConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA.format(level);
+      String rawDirQuota = Configuration.get(tierDirCapacityConf);
+      if (rawDirQuota.length() <= 0) {
+        System.err.format("Tier %d: Quota cannot be empty.%n", level);
+        return false;
+      }
+
+      String[] dirQuotas = rawDirQuota.split(",");
+
+      try {
+        Map<String, MountedStorage> storageMap = new HashMap<>();
+        File file = new File(dirPaths[0]);
+        if (dirPaths.length == 1 && alias.equals("MEM") && !file.exists()) {
```

Comment:
From what I understand our startup script only mount single RAM disk. If more than one directories are specified for the path of tier 0, then it's up to the user to mount the RAM disk separately. The purpose of this check is to skip checking only if we determine that the start up script will automatically mount the RAM disk.